# Titel

Gruppe: Gaia Capoferri, Dana Schnekenburger, Katarina Djuric

Fragen
- Welche Länder hatten in den letzten 50 Jahren die grösste Zunahme der Schuljahre?
- Was sind die Gründe dafür?

Quellen
- https://ourworldindata.org/grapher/mean-years-of-schooling-long-run?tab=chart

## 1. Daten beschaffen und laden

In [3]:
import pandas as pd
from matplotlib import pyplot as plt

In [4]:
# loading the data
df = pd.read_csv('mean-years-of-schooling-long-run/mean-years-of-schooling-long-run.csv', skiprows=1,
                 names=['Entity','Code','Year','Combined-average years of education for 15-64 years male and female youth and adults'])
df.head() 

Entity Code  Year  \
0  Afghanistan  AFG  1870   
1  Afghanistan  AFG  1875   
2  Afghanistan  AFG  1880   
3  Afghanistan  AFG  1885   
4  Afghanistan  AFG  1890   

   Combined-average years of education for 15-64 years male and female youth and adults  
0                                               0.01                                     
1                                               0.01                                     
2                                               0.01                                     
3                                               0.01                                     
4                                               0.01

## 2. Daten vorbereiten
- Die Roh-Daten sind bereits im Long-Format
- Für die Analyse werden die Daten pivotiert: Jede Spalte repräsentiert eine Region, die Zeilen sind die verschiedenen Jahre - in absteigender Reihenfolge von 5 Jahren.

In [5]:
df_pivot = df.pivot(index='Entity', columns='Year', values='Combined-average years of education for 15-64 years male and female youth and adults')
df_pivot.head()

Year            1870     1875      1880      1885      1890      1895  \
Entity                                                                  
Afghanistan  0.01000  0.01000  0.010000  0.010000  0.010000  0.010000   
Africa       0.05829  0.06893  0.084427  0.103117  0.120366  0.130437   
Albania      0.02000  0.03000  0.030000  0.040000  0.050000  0.060000   
Algeria      0.10000  0.13000  0.160000  0.200000  0.230000  0.270000   
Argentina    0.39000  0.49000  0.590000  0.650000  0.700000  0.830000   

Year             1900      1905      1910      1915  ...      1995      2000  \
Entity                                               ...                       
Afghanistan  0.010000  0.010000  0.010000  0.020000  ...  2.400000  3.030000   
Africa       0.148879  0.183318  0.233323  0.323424  ...  4.708579  5.012495   
Albania      0.080000  0.100000  0.150000  0.240000  ...  8.540000  9.930000   
Algeria      0.390000  0.510000  0.600000  0.580000  ...  5.760000  6.060000   
Argentina    0.970000  1.450000  1.810000  2.350000  ...  8.910000  9.080000   

Year              2005       2010      2015       2020      2025       2030  \
Entity                                                                        
Afghanistan   3.320000   3.930000  4.810000   5.690000   6.26000   6.840000   
Africa        5.446501   5.890455  6.195438   6.681866   7.18392   7.733528   
Albania      10.240000  10.440000  9.990000  10.320000  10.58000  10.830000   
Algeria       6.430000   7.000000  7.550000   8.180000   8.77000   9.370000   
Argentina     9.570000   9.710000  9.590000   9.860000  10.13000  10.390000   

Year              2035       2040  
Entity                             
Afghanistan   7.380000   7.920000  
Africa        8.256504   8.776776  
Albania      11.100000  11.390000  
Algeria      10.030000  10.590000  
Argentina    10.640000  10.870000  

[5 rows x 35 columns]

## 3. Statistik der Roh-Daten (Analyse in Long-Format)

über die einzelnen Spalten kann folgendes ausgesagt werden:
- Die Spalte *Entity* enthält 146 verschieden Entity 
- Die Spalte *Codes* enthält 146 verschiedene Codes
- Die Daten zwischen 1870 und 2040 sind vorhanden. Wir untersuchen nur von 1970 bis 2020
- Die Combined-average years of education for 15-64 years male and female youth and adults streut zwischen 1 und 1374

In [6]:
df.dtypes

Entity                                                                                   object
Code                                                                                     object
Year                                                                                      int64
Combined-average years of education for 15-64 years male and female youth and adults    float64
dtype: object

### 3.1 Analyse Entity

In [7]:
print('Anzahl verschiedener Entities: ', len(df['Entity'].unique()))

Anzahl verschiedener Entities:  153
